app/views/merchants/show.html.erb

```diff
@@ -0,0 +1,54 @@
+<div> 
+    <section>
+    <h1><%= @merchant.name %></h1>
+    <h2><%= link_to 'All Items', "/merchants/#{@merchant.id}/items" %> </h2>
+    <h2><%= link_to 'All Invoices', "/merchants/#{@merchant.id}/invoices" %> </h2>
+    </section>
+
+    <section>
+      <div>
+        <h1>Top 5 Customers</h1>
+        <table>
+          <tr> 
+            <th scope="col">Customer</th>
+            <th scope="col">Successful Transactions Total</th>
+          </tr>
+        </table>
+
+        <tbody>
+          <tr>
+            <th scope="row"></th>
+            <td></td>
+          </tr>
+        </tbody>
+      </div>
+    </section>
+
+    <section>
+      <div>
+        <h1> Items Ready to Ship </h1>
+        <table>
+          <tr> 
+            <th scope="col">Item</th>
+            <th scope="col">Invoice ID </th>
+            <th scope="col">Created On</th>
+          </tr>
+        </table>
+
+        <tbody class="shipment-table">
+        <% @shippable_items.each do |item| %>
+          <tr>
+            <th scope="row"><p><%= item.name %></p>
+            </th>
+            <td>
+              #<%= link_to "#{item.invoice_items.find_invoice_id(item.id)}", "/merchants/#{@merchant.id}/invoice/#{item.invoice_items.find_invoice_id(item.id)}" %>
```

Comment:
You can bypass referencing invoice items entirely because items know about their invoices. Therefore a `find_invoice_id` method can be created in the Item model which would clear a lot of this up.

Reply:
BLESS YOU, @jamogriff!!